hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/TestS3ADeleteOnExit.java

```diff
@@ -89,6 +90,17 @@ public void testDeleteOnExit() throws Exception {
     assertEquals(0, testFs.getDeleteOnDnExitCount());
   }
 
+  @Test
+  public void testCreateRequestFactoryWithInvalidChecksumAlgorithm() throws Exception {
+    Configuration conf = createConfiguration();
+    conf.set(Constants.CHECKSUM_ALGORITHM, "INVALID");
+    TestS3AFileSystem testFs  = new TestS3AFileSystem();
+    URI uri = URI.create(FS_S3A + "://" + BUCKET);
+    final IllegalArgumentException exception = intercept(IllegalArgumentException.class,
+        () -> testFs.initialize(uri, conf));
+    assertEquals("Checksum algorithm is not supported: INVALID", exception.getMessage());
```

Comment:
use AssertJ assertions in new tests...better assertions and makes the move to Junit5 easier